internal/provider/resource/secret_resource.go

```diff
@@ -219,14 +241,26 @@ func (r *secretResource) Create(ctx context.Context, req resource.CreateRequest,
 		// Set state to fully populated data
 		plan.WorkspaceId = types.StringValue(serviceTokenDetails.Workspace)
 	} else if r.client.Config.IsMachineIdentityAuth {
+		
+		// null check secret reminder
+		var secretReminderNote string
+		var secretReminderRepeatDays int64
+
+		if plan.SecretReminder != nil {
+			secretReminderNote = plan.SecretReminder.Note.ValueString()
+			secretReminderRepeatDays = plan.SecretReminder.RepeatDays.ValueInt64()
```

Comment:
Was there a reason you didn't add `plan.SecretReminder.RepeatDays.ValueInt64()` directly to `CreateRawSecretsV3`?

Reply:
Because the default value is 0 if null and the API accepts it, which shouldn't be allowed.

Reply:
We can also choose to handle this at the API level.

Reply:
Right, resolving so we don't have to make API changes